dev/src/timeout.rs

```diff
@@ -95,31 +98,68 @@ where
     })
 }
 
-pub struct LivenessGuard {
-    #[allow(unused)]
-    handle: JoinHandle<Result<()>>,
+struct LivenessGuardFuture<D: Debug> {
+    inner: Pin<Box<dyn Future<Output = D> + Send>>,
```

Comment:
Just to make sure I understand, this says that the struct LivenessGuardFuture has an inner field on the heap which can't be moved and implements a Future trait that's thread safe?

Reply:
Yep, you've got it.

Technically, it's a bit more specific: `Send` means that ownership can move between threads, but not a borrow to it. To share a borrow between threads, it would need `Sync`, too.